documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -1,7 +1,17 @@
 Pass Data to Algorithms
 #######################
 
-You can use one of the following ways to pass data to an algorithm executed with a device policy:
+When using C++ Standard Execution Policies, oneDPL supports data being passed to its algorithms as specified in the
+ISO/IEC 14882:2017 standard (commonly called C++17). According to this standard, it is the user's responsibility to
+avoid data races when using these parallel execution policies.
+
+Note: ``std::vector<bool>`` implementations are not required to avoid data races when the contents of the contained
+object in different elements in the same sequence are modified concurrently. Some implementations of
+``std::vector<bool>`` may optimize multiple ``bool`` elements into a bitfield, causing data races with parallel
+execution policies. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
+input with C++ Standard Execution Policies.
+
+When using a device execution policy, you can use one of the following ways to pass data to an algorithm:
 
 * ``oneapi:dpl::begin`` and ``oneapi::dpl::end`` functions
 * Unified shared memory (USM) pointers and ``std::vector`` with USM allocators
```

Comment:
In regard to ```std::vector`` with USM allocators`..
In spite of we have a comment 
`// for raw pointers and direct pass objects (for example, counting_iterator, iterator of USM-containers)`
I an afraid we don't (and cannot) differ  an iterator of the host data allocated container and an iterator of C++ container with USM allocator.

Reply:
In many cases I believe we can distinguish a iterator from a `std::vector` with `USM allocator`, however it is implementation dependent.
In #1438 I add some code which does to the extent that it is possible.  
Please see this comment for explanation:  https://github.com/oneapi-src/oneDPL/pull/1438#issuecomment-1989312899

I think it better to continue the discussion of this topic in that PR.  We could improve the documentation language there perhaps to be more conservative in our claims for USM allocators.